lib/sdk/core/observer.js

```diff
@@ -41,13 +41,12 @@ const ObserverDelegee = Class({
     this.delegate = delegate;
   },
   QueryInterface: function(iid) {
-    const isObserver = iid.equals(Ci.nsIObserver);
-    const isWeakReference = iid.equals(Ci.nsISupportsWeakReference);
+    if (iid.equals(Ci.nsIObserver) ||
+        iid.equals(Ci.nsISupportsWeakReference) ||
+        iid.equals(Ci.nsISupports))
+      return this;
```

Comment:
I think this organization will result in a strict warning about this function not returning.  Can we keep the old `if` `throw` `return` organization?